sql/log_event_server.cc

```diff
@@ -6211,7 +6232,8 @@ int Rows_log_event::do_apply_event(rpl_group_info *rgi)
       if (!table->in_use)
         table->in_use= thd;
 
-      error= do_exec_row(rgi);
+      error= do_exec_row(rgi, thd_info_cache);
+      THD_STAGE_INFO(thd, stage_executing);
```

Comment:
In addition, the `Rows_log_event::do_exec_row()`s previously reverts the state to `Executing` before they return.
I’ve moved that step here as well.

Curiously, while they changed `thd->wsrep_info` in WSRep to match, they didn’t “reset” it as well.